lib/src/widgets/yaru_page_indicator.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:flutter/material.dart';
+
+/// A Yaru page indicator.
+///
+/// If there's enough space, it will be rendered into a line of dots,
+/// if not, it will be rendered into a text-based indicator.
+class YaruPageIndicator extends StatelessWidget {
+  /// Create a [YaruPageIndicator].
+  const YaruPageIndicator({
+    super.key,
+    required this.length,
+    required this.page,
+    this.animationDuration = Duration.zero,
+    this.animationCurve = Curves.linear,
+    this.onDotTap,
+  }) : assert(page >= 0 && page <= length - 1);
+
+  /// Determine the number of pages.
+  final int length;
+
+  /// Current page index.
+  /// This value should be clamped between 0 and [length] - 1
+  final int page;
+
+  /// Duration of a transition between two dots.
+  /// Use [Duration.zero] (defaults) to disable transition.
+  final Duration animationDuration;
+
+  /// Curve used in a transition between two dots.
+  final Curve animationCurve;
+
+  /// Callback called when tapping a dot.
+  /// It passes the tapped page index as parameter.
+  final Function(int page)? onDotTap;
```

Comment:
could it be called "onTap"? "dot" is an unnecessary description of the style we have right now :)

```suggestion
  final ValueChanged<int>? onTap;
```
- https://api.flutter.dev/flutter/material/TabBar/onTap.html
- https://api.flutter.dev/flutter/material/BottomNavigationBar/onTap.html

Reply:
Well I disagree, because this indicator is a row of dot, and it can be also just a `Text()`.
`onDotTap` makes it obvious that it is a callback only for a single dot.
Imo `onTap` means this is a callback for the full indicator onTap event :)

Reply:
Sorry, I just saw your links 😅
You're right so!